documentation/book/security.adoc

```diff
@@ -0,0 +1,64 @@
+== Security
+
+The Apache Kafka project supports data encryption by means of the SSL/TLS protocol.
+This makes it possible to encrypt data transferred between brokers (interbroker communication) and between clients and brokers.
+Leveraging the SSL/TLS support, it is also possible to have mutual authentication, where the Kafka broker authenticates the client by verifying its certificate, and the client does likewise with the server's certificate.
+
+The Cluster Operator sets up the SSL/TLS certificates to provide this encryption and authentication.
+
+=== Certificates
+
+Each Kafka broker needs its own private and public keys in order to support encryption.
+The public key has to be signed by a certificate authority (CA) in order to have a related X.509 certificate for providing server authentication and encrypting the communication channel with the client (which could be another broker as well).
+More precisely, each broker has two different certificates (and related private keys), one for interbroker communication  and one for communication with clients.
+These broker certificates are signed using different CAs: The "internal-ca" signs the certificates used for interbroker communication, and the "clients-ca" signs the certificates used for client communication.
+The CAs themselves use self-signed certificates.
+The "internal-ca" is used across multiple clusters, that is, there is a single "internal-ca" for each instance of a cluster operator. 
```

Comment:
Really? I thought the original installation had it per cluster. Having it per CO is IMHO a risk. We should have at least an issue for fixing it.

Also, how does the internal CA work when it is in a different namespace than the Kafka cluster?

Reply:
I thought we agreed on having this way otherwise there were no reason for having the `AbstractAssemblyOperator` creating the internal-ca (which was something we agreed as well in the previous reviews); it's done in this way because it was not related to the specific cluster. Moving to be cluster specific means changing the current logic and having `KafkaCluster` handling this creation as all the other certificates.
I think that at this point an issue is better than changing this PR.

Reply:
Well, it cannot be part of `KafkaCluster`, because the first one using it will be `ZookeeperCluster`. So IMHO it needs to be created in `KafkaAssemblyOperator` or even in `AbstractAssemblyOperator`, but based on a cluster. But I don't think it should be CO wide. That would mean that the clusters are not secured against each other (if I have access to cluster `A` I can use the keys to authenticate to cluster `B`). 

I also didn't saw anywhere any handling of the namespace issue - that would be much easier with per cluster internal CA.

Reply:
Good point. I'm going to open an issue about that as soon as we'll merge this one. I'll work on this issue but maybe it's better for now merging a few stuff with CRD and Zk/Kafka log levels instead of adding code which generates more conflicts. It'll avoid to add pains on ... pains :-)